src/calcium/qqbar.jl

```diff
@@ -1485,16 +1485,21 @@ function ZZRingElem(a::QQBarFieldElem)
 end
 
 function (::Type{ComplexF64})(x::QQBarFieldElem)
-  z = AcbField(53, cached = false)(x)
-  x = ArbFieldElem()
-  ccall((:acb_get_real, libflint), Nothing, (Ref{ArbFieldElem}, Ref{AcbFieldElem}), x, z)
+  z = AcbField(57, cached = false)(x)
```

Comment:
@thofma so you set prec to 53 here (perhaps copied from ca.jl ?) In theory Float64 has 53 bits of mantissa, so it seems sensible to use 53 here. But in practice I don't think the mantissa size is quite the same thing as `prec`?

In any case, experimentally, I looked at `ComplexF64(sqrt(QQBar(-2))) - sqrt(Complex(-2))` and it turns out that for anything less than 57, I get a discrepancy there, meaning we loose precision.

Reply:
I wonder if @fredrik-johansson could perhaps illuminate this: what is the "best" & correct (!) way to get a `Float64` aka `double` approximation of the real / imaginary parts of a `qqbar` element? Same for `ca`.


I guess my question is: what precision do I need to pass to `qqbar_get_acb` resp. `qqbar_get_arb` ?

Reply:
@fredrik-johansson @thofma @fieker @albinahlback any of you have a hint for me as to how to correctly round an arb or acb into a Julia `Float64` / C `double`?

Reply:
Err, sorry, I mean `qqbar`

Reply:
Is it crucial to actually have correct rounding, or is ~1 unit in the last place acceptable? If you absolutely need correct rounding (to nearest), do ``qqbar_get_arb(y, x, 53+N)`` for some reasonable ``N`` and check ``arb_can_round_arf(y, 53, ARF_RND_NEAR)``; repeat with larger ``N`` if necessary. Note that this can be much slower than non-correct rounding. For a complex number you would need to apply ``arb_can_round_arf`` to both the real and imaginary parts.

Reply:
I don't think absolutely correct rounding is necessary there for most applications (we can provide that via a separate function if need be).

So how can I get ~1 unit in the last place?

Reply:
The example where I saw problems was `sqrt(-2)` but I just rechecked and the error in the imaginary part is indeed just one ULP. So that seems fine! OK I'll happily go back to 53 bits of precision then.